From ltsi-dev-bounces@lists.linuxfoundation.org Mon May 28 23:23:19 2012
From: "Shimoda, Yoshihiro" <yoshihiro.shimoda.uh@renesas.com>
Date: Tue, 29 May 2012 15:23:15 +0900
Subject: [LTSI-dev] [PATCH 08/12] usb: r8a66597-hcd: add function for external controller
To: ltsi-dev@lists.linuxfoundation.org
Message-ID: <4FC46B53.2070602@renesas.com>


>From 8eb5b6c291ca376adaff5cd123cfc39e186b30ee Mon Sep 17 00:00:00 2001
From: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
Date: Thu, 7 Jul 2011 09:57:10 +0900
Subject: usb: r8a66597-hcd: add function for external controller

R8A66597 has the pin of WR0 and WR1. So, if one write-pin of CPU
connects to the pins, we have to change the setting of FIFOSEL
register in the controller. If we don't change the setting,
the controller cannot send the data of odd length.

Signed-off-by: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
(cherry picked from commit f2e9039a43b01f01cab9dfaea2cad5f304fb3343)

Conflicts:

    drivers/usb/host/r8a66597-hcd.c
    drivers/usb/host/r8a66597.h

Signed-off-by: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
---
 include/linux/usb/r8a66597.h |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/include/linux/usb/r8a66597.h b/include/linux/usb/r8a66597.h
index 6e1bfae..7cd89e7 100644
--- a/include/linux/usb/r8a66597.h
+++ b/include/linux/usb/r8a66597.h
@@ -45,6 +45,9 @@ struct r8a66597_platdata {

 	/* set one = big endian, set zero = little endian */
 	unsigned	endian:1;
+
+	/* (external controller only) set one = WR0_N shorted to WR1_N */
+	unsigned	wr0_shorted_to_wr1:1;
 };

 /* Register definitions */
-- 
1.7.1
_______________________________________________
LTSI-dev mailing list
LTSI-dev@lists.linuxfoundation.org
https://lists.linuxfoundation.org/mailman/listinfo/ltsi-dev

